components/webFeatures.ts

```diff
@@ -115,24 +126,56 @@ webFeaturesRouter.get(
         if (!validateUserAndGv(req, res)) {
             return
         }
-
-        if (!req.query.level && !isNaN(parseInt(req.query.level))) {
-            res.json({ error: "bad level" })
+        if (!req.query.level) {
+            formErrorMessage(
+                res,
+                "The request must contain the level to set the escalation to.",
+            )
+            return
+        }
+        if (
+            isNaN(parseInt(req.query.level)) ||
+            parseInt(req.query.level) <= 0
+        ) {
+            formErrorMessage(res, "The level must be a positive integer.")
             return
         }
 
-        if (!req.query.id && !uuidRegex.test(req.query.id)) {
-            res.json({ error: "bad id" })
+        if (!req.query.id || !uuidRegex.test(req.query.id)) {
+            formErrorMessage(
+                res,
+                "The request must contain the uuid of an escalation.",
+            )
             return
         }
 
         try {
             await loadUserData(req.query.user, req.query.gv)
         } catch (e) {
-            res.json({ error: "failed to load user data" })
+            formErrorMessage(res, "Failed to load user data.")
             return
         }
-
+        if (controller.escalationMappings[req.query.id] === undefined) {
+            formErrorMessage(res, "Unknown escalation.")
+            return
+        }
+        if (
+            getLevelCount(controller.escalationMappings[req.query.id]) <
+            parseInt(req.query.level)
```

Comment:
```suggestion
            parseInt(req.query.level, 10)
```
Just in case any escalations come along that use double digits, need the proper radix.